packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -12,6 +14,46 @@ export interface AssetStagingProps extends FingerprintOptions {
    * The source file or directory to copy from.
    */
   readonly sourcePath: string;
+
+  /**
+   * Bundle the asset by executing a command in a Docker container.
+   * The asset path will be mounted at `/asset-input`. The Docker
+   * container is responsible for putting content at `/asset-output`.
+   * The content at `/asset-output` will be zipped and used as the
+   * final asset.
+   *
+   * @default - source is copied to staging directory
+   *
+   * @experimental
+   */
+  readonly bundling?: BundlingOptions;
+
+  /**
+   * How to calculate the hash for this asset.
+   *
+   * @default AssetHashType.SOURCE
+   */
+  readonly assetHashType?: AssetHashType;
```

Comment:
If we receive `CUSTOM` here because it was set like that in `s3-asset` or `lambda`, can/should `Staging.assetHash` just return the string `AssetHashType.CUSTOM` to avoid unnecessary fingerprint calculations?

Reply:
Feels like the custom hash should actually be propagated to the staging object so .assetHash represents the correct value. No?